lib/session_manager.rb

```diff
@@ -0,0 +1,77 @@
+require_relative './authenticator'
+require 'openssl'
+
+class SessionManager
+  attr_accessor :sessions, :authenticator
+
+  class WrongPassword < StandardError
+  end
+  class UnknownSession < StandardError
+  end
+
+  def initialize()
+    @sessions = {}
+    @authenticator = Authenticator.new()
+  end
+
+  def session_info(sess)
+    sess = sess.to_s
+    sessions[sess]
+  end
+
+  def signup(sess, user, salt, pass, payload)
+    sess = sess.to_s
+    @authenticator.register(user, salt, pass)
+    sessions[sess] = { user: user, login: true, token: nil, payload: payload }
+  end
+
+  def start_login(sess, user)
+    sess = sess.to_s
+    token = OpenSSL::BN.rand(256, 0, false).to_s(16)
```

Comment:
[nit] Ruby での乱数生成は OpenSSL 直接使うより [securerandom](https://docs.ruby-lang.org/ja/latest/class/SecureRandom.html) 使う方がやりやすいです。

```suggestion
    token = SecureRandom.hex(32)
```

Reply:
確かにこちらの方が簡潔ですね。安全な乱数発生のやり方が分からなかったので助かりました。